src/commands/deployAzurePipeline/configurePipeline.ts

```diff
@@ -0,0 +1,50 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from 'vscode-azureextensionui';
+
+export async function configurePipeline(context: IActionContext, target: any): Promise<void | undefined> {
+    const deployToAzureExtensionInstalled = await isDeployToAzureExtensionInstalled();
+    if (!deployToAzureExtensionInstalled) {
+        return undefined;
+    }
+
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    if (!cloudExplorer.available) {
+        return undefined;
+    }
+
+    const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+    if (clusterTarget && clusterTarget.cloudName === "Azure" && clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+        const configurePipelineCommand = 'configure-cicd-pipeline';
+        await executeDeployToAzureExtensionInstalled(configurePipelineCommand, clusterTarget);
+    } else {
+        vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        return undefined;
+    }
+
+}
+
+async function isDeployToAzureExtensionInstalled(): Promise<boolean> {
+    const deployToAzureExtensionId = 'ms-vscode-deploy-azure.azure-deploy';
+    const pipelinesExtension = vscode.extensions.getExtension(deployToAzureExtensionId);
+    if (!pipelinesExtension) {
+        vscode.window.showWarningMessage('Please install/enable `Deploy to Azure` extension and start again.');
+        await vscode.commands.executeCommand('extension.open', deployToAzureExtensionId);
```

Comment:
I'm not sure about doing this automatically.  I see that it gives the user some context for the message; but perhaps consider putting a button on the 'not installed' message saying "View in Marketplace" or "View in Extensions Tab" or something like that.

Reply:
``` await vscode.commands.executeCommand('extension.open', deployToAzureExtensionId); ```
This command only opens the extension in VSCode and doesn't install it automatically. 
@itowlson any particular concern you have around this?

Reply:
💡 Just adding my 2 cents as observation and merit to the proposed idea: (but it’s open for discussion)

Pro/advantage (button for user): If we follow the button approach: That makes user more aware about the actions they are taking I.e. for first time user it will display message below on right hand corner and followed by automatic market place and I wonder that will be just overwhelming for some user as experience.

Reply:
@kanika1894 Yes, I understand it only opens the extension, doesn't install it. I was just concerned about whether it was good to shift the user's focus in this way without further interaction, particularly since the visual shift will likely draw their attention more than the message which explains the visual shift.

Reply:
I guess @pulkitaggarwl might be able to better suggest in terms of expected user experience.

Reply:
Yeah, we can add a notification with a CTA "Open Extension". 
Here is the notification I am thinking -  "Deploy to Azure" VSCode extension is needed for this command. Please install/enable the extension and try again.  

@itowlson how does that look?

Reply:
![image](https://user-images.githubusercontent.com/58769601/95588097-9a7db580-0a60-11eb-9896-9f38b958e1e7.png)
Attaching screenshot for reference.

Reply:
Thanks minor consistency nit: see here : https://github.com/Azure/vscode-aks-tools/pull/42/files#diff-e6f9b94f8c32fa85137a98cfd4e5f259R46 How you have link to the "Deploy to Azure" we could do same link here i.e. `(Deploy to Azure)[link]` Just an Idea.

**Approved** from my end. Passing on for Ivan's eyes for a last take. Thank you so much to everyone.

Reply:
@itowlson suggested "Show extension" instead of "Open". Have updated the same.